glusterd2/commands/volumes/volume-option.go

```diff
@@ -107,11 +167,15 @@ func volumeOptionsHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	txn.Steps = []*transaction.Step{
-		lock,
 		{
 			DoFunc: "vol-option.Validate",
 			Nodes:  []uuid.UUID{gdctx.MyUUID},
 		},
+		{
+			DoFunc: "vol-option.XlatorAction",
```

Comment:
This needs UndoFunc as well. ActorFuncs can do changes that will need to be undone on any failure.

Reply:
@kshlm Done